src/core/providers/ogr/qgsogrproviderutils.cpp

```diff
@@ -719,6 +720,23 @@ QStringList QgsOgrProviderUtils::wildcards()
   return createFilters( QStringLiteral( "wildcards" ) ).split( '|' );
 }
 
+QStringList QgsOgrProviderUtils::tableNamesFromSelectSQL( const QString &sql )
+{
+  QStringList tableNames;
+  const QgsSQLStatement statement { sql };
+  const QgsSQLStatement::NodeSelect *nodeSelect { static_cast<const QgsSQLStatement::NodeSelect *>( statement.rootNode() ) };
```

Comment:
Are we guaranteed that the root node is a nodeselect? Maybe a dynamic_cast is safer here, given the check on line 728:

```suggestion
  const QgsSQLStatement::NodeSelect *nodeSelect { dynamic_cast<const QgsSQLStatement::NodeSelect *>( statement.rootNode() ) };
```